consul/install.go

```diff
@@ -0,0 +1,78 @@
+package consul
+
+import (
+	"fmt"
+	"path"
+
+	"github.com/layer5io/gokit/errors"
+
+	"github.com/layer5io/meshery-consul/internal/config"
+
+	"github.com/layer5io/meshery-adapter-library/adapter"
+)
+
+type MeshInstance struct{}
+
+func (m *MeshInstance) applyUsingManifest(request adapter.OperationRequest, operation *adapter.Operation, h *Handler) error {
+	err := h.ApplyKubernetesManifest(request, *operation, map[string]string{"namespace": request.Namespace},
+		path.Join("consul", "config_templates", operation.Properties[config.OperationTemplateNameKey]))
+	return err
+}
+
+func (m *MeshInstance) getServicePorts(request adapter.OperationRequest, operation *adapter.Operation, h *Handler) ([]int64, error) {
+	return h.GetServicePorts(operation.Properties[config.OperationServiceNameKey], request.Namespace)
+}
+
+func (h *Handler) applyUsingManifest(request adapter.OperationRequest, operation *adapter.Operation) (string, error) {
+	status := "installing"
+
+	if request.IsDeleteOperation {
+		status = "removing"
+	}
+
+	meshInstance := &MeshInstance{}
+
+	err := h.Config.MeshInstance(meshInstance)
+	if err != nil {
+		return status, adapter.ErrMeshConfig(err)
+	}
+
+	h.Log.Info(fmt.Sprintf("%s %s", status, operation.Properties[config.OperationDescriptionKey]))
+	err = meshInstance.applyUsingManifest(request, operation, h)
+	if err != nil {
+		h.Log.Err(fmt.Sprintf("%s %s failed", status, operation.Properties[config.OperationDescriptionKey]), adapter.ErrInstallMesh(err).Error())
+		return status, adapter.ErrInstallMesh(err)
+	}
+
+	return "deployed", nil
+}
+
+func (h *Handler) getServicePorts(request adapter.OperationRequest, operation *adapter.Operation) (string, []int64, error) {
+	meshInstance := &MeshInstance{}
+
+	err := h.Config.MeshInstance(meshInstance)
+	if err != nil {
+		return "", nil, adapter.ErrMeshConfig(err)
+	}
+
+	svc := operation.Properties[config.OperationServiceNameKey]
+
+	h.Log.Info(fmt.Sprintf("Retreiving service ports for service %s.", svc))
+
+	var ports []int64
+	ports, err = meshInstance.getServicePorts(request, operation, h)
+	if err != nil {
+		error := errors.New("0000", fmt.Sprintf("Unable to retrieve information from the mesh: %s.", err.Error()))
```

Comment:
`0000` should this be defined as a const error code?

Reply:
Yes, agreed. I used it as a placeholder code for later discussion, but should have created an issue for this. 
The reason for this code was that I couldn't find one here https://github.com/layer5io/meshkit/blob/master/errors/codes.go (and consequently here https://github.com/mgfeller/meshery-adapter-library/blob/master/adapter/error.go) that I thought fit. 

The error occurs when, for some reason, information cannot be retrieved. 

Do you think one of the existing ones fits?

What could be a good name for a new error that reflects this situation?

ErrGetInfo, ErrGetData?

It could be tempting to have a general error that can be used if none fits. Then at least the same general error would be used everywhere and could more easily be found and replaced.